### DISCUSSION TITLE
Serial to I2s

### DISCUSSION THREAD

**Discussion by CFTechno** [pschatzmann/arduino-audio-tools#1190] [Q&A] [answered]
I have audio coming in over serial and would like it to be forwarded to a speaker and to a ble serial port. From the documentation i understand that the "audio source" can be any class implementing streaming so serial should work. I do fail to understand how to define this "in" stream. Any pointers?

**Comment by pschatzmann** [comment, marked as answer]
You are thinking too complicated: the standard Arduino Serial object is implementing Stream, so that's what you would use!
I suggest that you have a look at the corresponding examples that can be found in the [examples-communication/serial](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication/serial) folder

**Comment by CFTechno** [comment]
Thank you, seems i somehow failed to find this example https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-communication/serial/send-8bit-receive/send-8bit-receive.ino
